lib/parse.ts

```diff
@@ -25,6 +37,15 @@ export const parseFile = (file: File | string): Promise<UsersData> =>
           reject(new Error('No user rows found'))
           return
         }
+        const invalidRows = rows.reduce<[number, UserRow][]>((current, row, index) => {
+          if (!row.email.match(emailRegEx)) {
+            return [...current, [index, row]]
+          }
+          return current
+        }, [])
```

Comment:
To make this easier to read, let's make it `filter` and then `map`, not `reduce`.

Reply:
> To make this easier to read, let's make it `filter` and then `map`, not `reduce`.

If we use `filter` at first, then it's a bit tricky to save the invalid row indexes

Reply:
Does this work: https://github.com/zilinj88/nextjs-mailmerge/pull/14/commits/d4b31189690e86e0973575455e658a61aa552049